subsys/bluetooth/controller/ll_sw/ull_llcp_phy.c

```diff
@@ -0,0 +1,1118 @@
+enum {
+	/* Procedure run */
+	RP_PU_EVT_RUN,
+
+	/* Request recieved */
+	RP_PU_EVT_PHY_REQ,
+
+	/* Ack received */
+	RP_PU_EVT_ACK,
+
+	/* Indication recieved */
+	RP_PU_EVT_PHY_UPDATE_IND,
+};
+
+/* Hardcoded instant delta +6 */
+#define PHY_UPDATE_INSTANT_DELTA 6
+
+#if defined(CONFIG_BT_CENTRAL)
+/* PHY preference order*/
+#define PHY_PREF_1 PHY_2M
+#define PHY_PREF_2 PHY_1M
+#define PHY_PREF_3 PHY_CODED
+
+static inline uint8_t pu_select_phy(uint8_t phys)
+{
+	/* select only one phy, select prefered */
+	if (phys & PHY_PREF_1) {
+		return PHY_PREF_1;
+	} else if (phys & PHY_PREF_2) {
+		return PHY_PREF_2;
+	} else if (phys & PHY_PREF_3) {
+		return PHY_PREF_3;
+	} else {
+		return 0U;
+	}
+}
+
+static void pu_prep_update_ind(struct ll_conn *conn, struct proc_ctx *ctx)
+{
+	ctx->data.pu.tx = pu_select_phy(ctx->data.pu.tx);
+	ctx->data.pu.rx = pu_select_phy(ctx->data.pu.rx);
+
+	if (ctx->data.pu.tx != conn->lll.phy_tx) {
+		ctx->data.pu.c_to_p_phy = ctx->data.pu.tx;
+	} else {
+		ctx->data.pu.c_to_p_phy = 0U;
+	}
+	if (ctx->data.pu.rx != conn->lll.phy_rx) {
+		ctx->data.pu.p_to_c_phy = ctx->data.pu.rx;
+	} else {
+		ctx->data.pu.p_to_c_phy = 0U;
+	}
+}
+#endif /* CONFIG_BT_CENTRAL */
+
+#if defined(CONFIG_BT_PERIPHERAL)
+static uint8_t pu_select_phy_timing_restrict(struct ll_conn *conn, uint8_t phy_tx)
+{
+	/* select the probable PHY with longest Tx time, which
+	 * will be restricted to fit current
+	 * connEffectiveMaxTxTime.
+	 */
+	/* Note - entry 0 in table is unused, so 0 on purpose */
+	uint8_t phy_tx_time[8] = { 0,	      PHY_1M,	 PHY_2M,    PHY_1M,
+				   PHY_CODED, PHY_CODED, PHY_CODED, PHY_CODED };
+	struct lll_conn *lll = &conn->lll;
+	const uint8_t phys = phy_tx | lll->phy_tx;
+
+	return phy_tx_time[phys];
+}
+#endif /* CONFIG_BT_PERIPHERAL */
+
+static void pu_set_timing_restrict(struct ll_conn *conn, uint8_t phy_tx)
+{
+	struct lll_conn *lll = &conn->lll;
+
+	lll->phy_tx_time = phy_tx;
+}
+
+static void pu_reset_timing_restrict(struct ll_conn *conn)
+{
+	pu_set_timing_restrict(conn, conn->lll.phy_tx);
+}
+
+static uint16_t pu_event_counter(struct ll_conn *conn)
+{
+	struct lll_conn *lll;
+	uint16_t event_counter;
+
+	/* TODO(thoh): Lazy hardcoded */
+	uint16_t lazy = 0;
+
+	/**/
+	lll = &conn->lll;
+
+	/* Calculate current event counter */
+	event_counter = lll->event_counter + lll->latency_prepare + lazy;
+
+	return event_counter;
+}
+
+#if defined(CONFIG_BT_PERIPHERAL)
+static uint8_t pu_check_update_ind(struct ll_conn *conn, struct proc_ctx *ctx)
+{
+	/* Both tx and rx PHY unchanged */
+	if (!((ctx->data.pu.c_to_p_phy | ctx->data.pu.p_to_c_phy) & 0x07)) {
+		/* if no phy changes, quit procedure, and possibly signal host */
+		ctx->data.pu.error = BT_HCI_ERR_SUCCESS;
+		return 1;
+	} else {
+		/* if instant already passed, quit procedure with error */
+		if (is_instant_reached_or_passed(ctx->data.pu.instant, pu_event_counter(conn))) {
+			ctx->data.pu.error = BT_HCI_ERR_INSTANT_PASSED;
+			return 1;
+		}
+	}
+	return 0;
+}
```

Comment:
I suppose 0 == Success, and 1 == failure;

It feels like an uncommon way of handling error values, due to it being a unsigned value and also that the error value is undefined. 

Possibly refactor to return int and use errno or use a boolean value instead

Reply:
This code also causes CI failure and is refactored